python/test/ngraph/test_reduction.py

```diff
@@ -42,3 +42,38 @@ def test_reduction_ops(ng_api_helper, numpy_function, reduction_axes):
     expected = numpy_function(input_data, axis=reduction_axes)
     result = run_op_node([input_data], ng_api_helper, reduction_axes)
     assert np.allclose(result, expected)
+
+
+def test_reduce():
+    from functools import reduce
+    np.random.seed(133391)
+
+    # default reduce all axes
+    init_val = np.float32(0.)
+    input_data = np.random.randn(3, 4, 5).astype(np.float32)
+    expected = np.sum(input_data)
+    reduction_function_args = [init_val, ng.impl.op.Add]
+    result = run_op_node([input_data], ng.reduce, *reduction_function_args)
+    assert np.allclose(result, expected)
+
+    reduction_axes = (0, 2)
+    init_val = np.float32(0.)
+    input_data = np.random.randn(3, 4, 5).astype(np.float32)
+    expected = np.sum(input_data, axis=reduction_axes)
+    reduction_function_args = [init_val, ng.impl.op.Add, list(reduction_axes)]
+    result = run_op_node([input_data], ng.reduce, *reduction_function_args)
+    assert np.allclose(result, expected)
+
+    reduction_axes = (0, )
+    input_data = np.random.randn(100).astype(np.float32)
+    expected = reduce(lambda x, y: x - y, input_data, np.float32(0.))
```

Comment:
Having just written the wall of text above, I now see that you are doing the wrapping here. :) (Though I think the order of the `node` and `reduction_node` is flipped, and the docstrings above are incorrect---what you have here isn't really a `Node`, but a function on two `Node`s that produces a `Node`.) That should work just fine, and if it feels more "Pythonic" (I'm not a Python expert by the remotest stretch of the imagination) I think it makes sense. Would probably be good to also have a variant that takes an nGraph `Function` object in place of the Python-level `lambda`.